### PR TITLE
Drive: share_file, move_file_to_folder, download_file, export_pdf, touch_file

### PR DESCRIPTION
**Pull request by RajeshGogo**
# Description
Drive: share_file, move_file_to_folder, download_file, export_pdf, touch_file

Fixes # (230522234)

## Has it been tested?
- [X] Development testing done
- [ ] Unit or integration test implemented

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have performed a peer-reviewed with team member(s)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: drive/snippets/drive_v2/file snippets/download_file.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  const dest = fs.createWriteStream('/tmp/photo.jpg');
+  fileId = realFileId;
+  const buffers = [];
+  try {
+    await service.files.get({
+      fileId: fileId,
+      alt: 'media',
+    }, {responseType: 'stream'},
+    function( res) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/download_file.js:41]: Why is this mixing styles? Should either be callback or promise/async-await but not both at the same time.

File: drive/snippets/drive_v2/file snippets/export_pdf.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const dest = fs.createWriteStream('resume.pdf');
+  fileId = realFileId;
+  const buffers = [];
+
+  try {
+    await service.files.export({
+      fileId: fileId,
+      mimeType: 'application/pdf',
+    }, {responseType: 'stream'},
+    function(res) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/export_pdf.js:41]: Same as other files --- either use callback or promise, but not both

File: drive/snippets/drive_v2/file snippets/export_pdf.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          })
+          .on('end', function() {
+            console.log('Done');
+            Buffer.concat(buffers);
+          })
+          .pipe(dest);
+    },
+    );
+  } catch (err) {
+    throw err;
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/export_pdf.js:54]: Add //TODO

File: drive/snippets/drive_v2/file snippets/download_file.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          })
+          .on('end', function() {
+            console.log('Done');
+            Buffer.concat(buffers);
+          })
+          .pipe(dest);
+    },
+    );
+  } catch (err) {
+    throw err;
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/download_file.js:54]: Add // TODO

File: drive/snippets/drive_v2/file snippets/share_file.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  try {
+    const res = await service.permissions.insert({
+      resource: permission,
+      fileId: fileId,
+      fields: 'id',
+    });
+    console.log('Permission ID:', res.data.id);
+    ids.push(res.id);
+  } catch (err) {
+    throw err;
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/share_file.js:59]: add // TODO

File: drive/snippets/drive_v2/file snippets/download_file.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  }
+};
+// [END drive_download_file]
+
+module.exports = downloadFile;
+if (module=== require.main) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v2/file snippets/download_file.js:60]: A few formatting/indentation quirks throughout these files. Reformat for consistency.

**Review by sqrrrl — CHANGES_REQUESTED** [review]

File: drive/snippets/drive_v2/file snippets/export_pdf.js (hunk 2 above)
Comment (anuraggoogler) [reply drive/snippets/drive_v2/file snippets/export_pdf.js]: Done

File: drive/snippets/drive_v2/file snippets/export_pdf.js (hunk 3 above)
Comment (anuraggoogler) [reply drive/snippets/drive_v2/file snippets/export_pdf.js:43]: Done

File: drive/snippets/drive_v2/file snippets/download_file.js (hunk 4 above)
Comment (anuraggoogler) [reply drive/snippets/drive_v2/file snippets/download_file.js:43]: Done

File: drive/snippets/drive_v2/file snippets/download_file.js (hunk 6 above)
Comment (anuraggoogler) [reply drive/snippets/drive_v2/file snippets/download_file.js]: Done

**Pull request merged**